prometheus/histogram.go

```diff
@@ -1835,3 +1835,143 @@ func (n *nativeExemplars) addExemplar(e *dto.Exemplar) {
 		n.exemplars = append(n.exemplars[:nIdx], append([]*dto.Exemplar{e}, append(n.exemplars[nIdx:rIdx], n.exemplars[rIdx+1:]...)...)...)
 	}
 }
+
+type constNativeHistogram struct {
+	desc *Desc
+	dto.Histogram
+	labelPairs []*dto.LabelPair
+}
+
+func validateCount(count uint64, negativeBuckets, positiveBuckets map[int]int64, zeroBucket uint64) error {
+	var bucktCountSum int64
+	for _, v := range positiveBuckets {
+		bucktCountSum += v
+	}
+	for _, v := range negativeBuckets {
+		bucktCountSum += v
+	}
+	bucktCountSum += int64(zeroBucket)
+	if bucktCountSum != int64(count) {
+		return fmt.Errorf("bucket sums do not add up to total count")
+	}
+	return nil
+}
+
+func NewConstNativeHistogram(
```

Comment:
Needs doc comment.

Reply:
We should also have a version `MustNewConstNativeHistogram` (which wraps `NewConstNativeHistogram`, panicking where `NewConstNativeHistogram` would have returned an error).

Reply:
Added the docstring